hw06-reflection/src/main/java/ru/otus/TestRunner.java

```diff
@@ -0,0 +1,79 @@
+package ru.otus;
+
+import ru.otus.annotations.After;
+import ru.otus.annotations.Before;
+import ru.otus.annotations.Test;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class TestRunner {
+    private Class<?> clazz;
+    private List<Method> beforeMethods = new ArrayList();
+    private List<Method> afterMethods = new ArrayList();
+    private List<Method> testMethods = new ArrayList();
+
+    public TestRunner(String testClass) throws ClassNotFoundException, IllegalArgumentException {
+        clazz = Class.forName(testClass);
+        if (clazz.getAnnotationsByType(Test.class).length == 0) {
+            throw new IllegalArgumentException(String.format("%s has no @Test annotation", clazz));
+        }
+
+        for (Method method: clazz.getMethods()) {
+            updateMethods(Before.class, method, beforeMethods);
+            updateMethods(After.class, method, afterMethods);
+            updateMethods(Test.class, method, testMethods);
+        }
+        if (testMethods.size() == 0) {
```

Comment:
```suggestion
        if (testMethods.isEmpty()) {
```

В общем случае `isEmpty()` будет эффективнее, тк вам нужно проверить на пустоту, а не посчитать количество.
Коллекция сама знает, как эффективнее сделать эту проверку.
С `ArrayList` никакой разницы не будет, но с другими коллекциями может стать заметным.
К тому же так код становится понятнее.